this is a long unsigned int, initialising it with a pointer doesn't make sense
--
diff --git a/procinterrupts.c b/procinterrupts.c
index e91b203..a60fe00 100644
--- a/procinterrupts.c
+++ b/procinterrupts.c
@@ -107,10 +107,10 @@ static void guess_arm_irq_hints(char *name, struct irq_info *info)
 	static int compiled = 0;
 	/* Note: Last entry is a catchall */
 	static struct irq_match matches[] = {
-		{ "eth.*" ,{NULL} ,NULL, IRQ_TYPE_LEGACY, IRQ_GBETH },
-		{ "[A-Z0-9]{4}[0-9a-f]{4}", {NULL} ,check_platform_device, IRQ_TYPE_LEGACY, IRQ_OTHER},
-		{ "PNP[0-9a-f]{4}", {NULL} ,check_platform_device, IRQ_TYPE_LEGACY, IRQ_OTHER},
-		{ ".*", {NULL}, NULL, IRQ_TYPE_LEGACY, IRQ_OTHER},
+		{ "eth.*" ,{0} ,NULL, IRQ_TYPE_LEGACY, IRQ_GBETH },
+		{ "[A-Z0-9]{4}[0-9a-f]{4}", {0},check_platform_device, IRQ_TYPE_LEGACY, IRQ_OTHER},
+		{ "PNP[0-9a-f]{4}", {0},check_platform_device, IRQ_TYPE_LEGACY, IRQ_OTHER},
+		{ ".*", {0}, NULL, IRQ_TYPE_LEGACY, IRQ_OTHER},
 		{NULL},
 	};
 
